instrumentation/vertx-web-3.0/src/main/java/io/opentelemetry/javaagent/instrumentation/hypertrace/vertx/HttpRequestInstrumentation.java

```diff
@@ -68,7 +68,10 @@
         HttpRequestInstrumentation.class.getName() + "$WriteRequestAdvice_string");
     transformers.put(
         isMethod()
-            .and(named("write").and(takesArguments(1)).and(takesArgument(0, is(Buffer.class)))),
+            .and(
+                named("write")
+                    .and(takesArguments(1))
+                    .and(takesArgument(0, named("io.vertx.core.buffer.Buffer")))),
```

Comment:
why this change? I would prefer to keep it as a type instead of a string.

Reply:
When I removed the vertx dependency from the `runtimeClasspath`, the `byteBuddyJava` task starts failing with a `java.lang.NoClassDefFoundError: io/vertx/core/buffer/Buffer`. This surprised me because I assumed that this task leveraged the `compileClasspath` rather than the `runtimeClasspath`, but perhaps it builds its own classpath derived from the `runtimeClasspath`? When I went looking at other projects in this repo, I noticed that this project was the only one to use the class literals in the `transformers`, which is why the other projects didn't have a problem with removing the library being instrumented from the `runtimeClasspath`.


I too would prefer the type instead of the `String`. Writing up this comment made me realize that the reason removing vertx from the `runtimeClasspath` causes the `byteBuddyJava` task to fa
il is because we scope the `ByteBuddyPluginConfigurator` to only use the `runtimeClasspath` as shown here:

```kotlin
afterEvaluate{
    io.opentelemetry.instrumentation.gradle.bytebuddy.ByteBuddyPluginConfigurator(project,
            sourceSets.main.get(),
            "io.opentelemetry.javaagent.tooling.muzzle.collector.MuzzleCodeGenerationPlugin",
            project(":javaagent-tooling").configurations["instrumentationMuzzle"] + configurations.runtimeClasspath
    ).configure()
}
```

If instead we use `configurations.compileClasspath` rather than `configurations.runtimeClasspath` to configure the classpath for the bytebuddy gradle plugin. But, I'm not sure if that will have any other negative impacts. I looked closer at the open telemetry java nstrumentation project and saw that they also restricted the byteBuddyJava classpath to the tooling and runtime classpaths. As a result, the parts of their code that execute during the `byteBuddyJava` class use string literals for class names rather than the actual types, such as with their [Vertx Web HttpRequestInstrumentation class](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/0b0516bd1a0599c8f536cc6b2151782000681f39/instrumentation/vertx-web-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vertx/client/HttpRequestInstrumentation.java#L48)

Reply:
Following up, while I too would prefer to use the type rather than the `String` name, I also am weary of configuring bytebuddy to use the compileClasspath. As a result, I think we should follow the standard set by otel and keep these as Strings

Reply:
sounds good, thanks for the explanation :)